File: solutions/palindromeNumber-9/palindromeNumber-9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:

        if x<0:
            return False

        if x>=0 and x<=9:
            return True

        result = 0
        temp = x
        while temp>=10 :
            result = result*10 + temp%10
            temp = temp//10
        result = result*10 + temp%10

        if result == x:
            return True
        else:
            return False

File: solutions/palindromeNumber-9/palindromeNumber-9.cpp

/**********************************************************************************
*
* Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward.
*
* Input: s = 121
* Output: true
*
*
**********************************************************************************/

class Solution {

  /*
  *   [Revert half of the number]
  *
  *   Time complexity : O(log10(n)). We divided the input by 10 for every iteration, so the time complexity is O(log10(n)).
  *
  *   Space complexity : O(1).
  */

public:
    bool isPalindrome(int x) {

        if (x<0 || (x%10 == 0 && x!=0)) {
            return false;
        }

        int result = 0;
        while (x > result) {
            result = result*10 + x%10;
            x /= 10;
        }
        return x==result || x == result/10;
    }
};
